internal/pubsub/drivertest/drivertest.go

```diff
@@ -193,23 +196,29 @@ func testSendReceive(t *testing.T, newHarness HarnessMaker) {
 		if err := top.Send(ctx, m); err != nil {
 			t.Fatal(err)
 		}
-		want = append(want, m)
+		want = append(want, Message{
+			Body:     string(m.Body),
+			Metadata: m.Metadata,
+		})
 	}
 
 	// Receive from the subscription.
-	var got []*pubsub.Message
+	var got []Message
 	for i := 0; i < len(want); i++ {
 		m, err := sub.Receive(ctx)
 		if err != nil {
 			t.Fatal(err)
 		}
-		got = append(got, m)
+		got = append(got, Message{
+			Body:     string(m.Body),
+			Metadata: m.Metadata,
+		})
 		m.Ack()
 	}
 
 	// Check that the received messages match the sent ones.
-	less := func(x, y *pubsub.Message) bool { return bytes.Compare(x.Body, y.Body) < 0 }
-	if diff := cmp.Diff(got, want, cmpopts.SortSlices(less), cmpopts.IgnoreUnexported(pubsub.Message{})); diff != "" {
+	less := func(x, y Message) bool { return x.Body < y.Body }
```

Comment:
WDYT about adding a `String() string` func to `Message instead?

Reply:
That works. Using that for now.

Reply:
Okay, based on our chat offline, I'm going to see if I can add a `String()` method to `pubsub.Message` that gives readable output and also deals with the `Body` field possibly being non-UTF-8.

Reply:
Adding a String method to pubsub.Message has no effect on cmp.Diff, so that's a dead-end.

The root of the problem is all these unexported fields on Message. We can't easily get rid of them because so many things depend on them, but we can make a new type called say MessageValue that contains them and add a method to convert Message to MessageValue for testing purposes. Making a new PR for that: https://github.com/google/go-cloud/pull/1014.